src/Gui/Utilities.h

```diff
@@ -441,6 +441,30 @@ class GuiExport ItemViewSelection
     class MatchName;
 };
 
+#define FC_ADD_ENTRY(__part__, __partclass__, __parent__) SO_KIT_ADD_CATALOG_ENTRY(__part__, __partclass__, TRUE, __parent__, "", TRUE);
+
+#define FC_SET_SWITCH(__name__, __state__) \
+do { \
+    SoSwitch* sw = SO_GET_ANY_PART(this, __name__, SoSwitch); \
+    sw->whichChild = __state__; \
+} while (0)
+
+[[nodiscard]] inline auto getRotationComponents(const SbRotation &rotation) {

```

Comment:
Should use dedicated struct:

```
struct RotationComponents {
    float angle;
    SbVec3f axis;
}
```

Then you can access members via names, not indexes.

Reply:
fixed